vpr/src/place/analytic_placer.h

```diff
@@ -0,0 +1,249 @@
+ *
+ **************************************************************************************************************
+ * 											Algorithm Overview												  *
+ **************************************************************************************************************
+ *
+ * The most common objective function for placement is the sum of half-perimeter wirelengths (HPWL) over all nets.
+ * Efficient AP techniques approximate this objective function with a function that can be minimized efficiently.
+ *
+ * First, all multi-pin nets are converted into a set of 2-pin connections. In SimPL/HeAP, the Bound2bound net
+ * model is used. For each multi-pin net, the blocks with the minimum and maximum locations on a net (so-called
+ * bound-blocks) are connected to each other and to each internal block on the net. In other words, for a p-terminal
+ * net, each internal block has 2 connections, one to each bound block, and each bound block has p-1 connections,
+ * one to every block other than itself.
+ *
+ * Then, the weighted sum of the squared lengths of these 2-pin connections are minimized. This objective function
+ * can be separated into x and y components and cast in matrix form. To minimize this degree-2 polynomial, partial
+ * derivative is taken with respect to each variable. Setting the resulting system of linear equations to 0 gives
+ * the following equation (only x direction shown):
+ * 											Qx = -c
+ * where Q is a matrix capturing all connection between movable objects (objects to solve), x is a vector of all
+ * block locations, and c is a vector representing connections between movable and fixed objects.
+ * *** for detailed derivation and an example, refer to HeAP paper cited below.
+ *
+ * After formulating Q and c, a standard off-the-shelf solver (Eigen package) is used to solve for x. This completes
+ * the lower-bound placement.
+ *
+ * However, since the objective function does not take placement constraints into consideration, the generated
+ * solution is not legal. It generally has many blocks overlapping with one another, and the blocks may be on
+ * incompatible physical tiles. To legalize this solution, a geometric partitioning and spreading technique, introduced
+ * in SimPL, is used (@see cut_sreader.h). This completes the upper-bound placement.
+ *
+ * After the completion of 1 iteration of lower-bound & upper-bound placement, artificial pseudo connections are created
+ * between each block and its target location in the legalized overlap-free placement. When the mathematical system is
+ * again formulated and solved, the pseudo connections pull blocks towards their target locations, which tends to reduce
+ * overlaps in the placement.
+ *
+ * This process of formulating the system, solving, and legalizing is repeated until sufficiently good placement is
+ * acquired. Currently the stopping criterion is 15 iterations without improvement in total_hpwl.
```

Comment:
Change 15 to your new defined constant name. (you can say 15 at time of this writing if you like too, but good to link to the constant name).